code/game/machinery/doors/door.dm

```diff
@@ -76,8 +76,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
+		if(!M.restrained())
+			if(ispet(AM))

```

Comment:
You should probably be checking `M` instead of `AM` here, even though it functionally makes no difference

Reply:
This code prevents "pet" mobs that were big enough to bump-open airlocks from doing so, if they lacked a collar. (I'm not sure there were any "pet" mobs that were larger than `MOB_SIZE_SMALL` though)

Reply:
@Crazylemon64 This code checks if there is a pet that has a collar, bumpopen the door. If they aren't a pet, then check their mob size, and is they are bigger than small.. bumpopen. If I reversed this.. pets wouldn't be able to open doors at all because of the size check.

Reply:
Wait.. I misunderstood you @Crazylemon64
Hmm